app/assets/javascripts/govuk_publishing_components/load-analytics.js

```diff
@@ -71,38 +73,119 @@ window.GOVUK.loadAnalytics = {
     }
   },
 
+  loadGa4Vars: function () {
+    // If we recognise the environment, load the GA4 vars
+    if (this.ga4EnvironmentVariables[this.environment]) {
+      window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+      window.GOVUK.analyticsGa4.vars = window.GOVUK.analyticsGa4.vars || {}
+      window.GOVUK.analyticsGa4.vars.id = 'GTM-MG7HG5W'
+      window.GOVUK.analyticsGa4.vars.auth = this.ga4EnvironmentVariables[this.environment].auth
+      window.GOVUK.analyticsGa4.vars.preview = this.ga4EnvironmentVariables[this.environment].preview
+      window.GOVUK.analyticsGa4.vars.environment = this.environment // Used for testing and debugging
+
+      window.GOVUK.analyticsGa4.vars.gem_version = 'not found'
+      var gemMeta = document.querySelector('meta[name="govuk:components_gem_version"]')
+      if (gemMeta) {
+        window.GOVUK.analyticsGa4.vars.gem_version = gemMeta.getAttribute('content')
+      }
+
+      window.GOVUK.analyticsGa4.vars.internalDomains = []
+      window.GOVUK.analyticsGa4.vars.internalDomains.push(window.GOVUK.analyticsGa4.core.trackFunctions.getHostname())
+      window.GOVUK.analyticsGa4.core.trackFunctions.appendDomainsWithoutWWW(window.GOVUK.analyticsGa4.vars.internalDomains)
+    }
+  },
+
+  addGtmScriptToDom: function () {
+    var firstScript = document.getElementsByTagName('script')[0]
+    var newScript = document.createElement('script')
+    newScript.async = true
+
+    // initialise GTM
+    window.dataLayer = window.dataLayer || []
+    window.dataLayer.push({ 'gtm.blocklist': ['customPixels', 'customScripts', 'html', 'nonGoogleScripts'] })
+    window.dataLayer.push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' })
+
+    var auth = window.GOVUK.analyticsGa4.vars.auth || ''
+    var preview = window.GOVUK.analyticsGa4.vars.preview || ''
+    if (auth) {
+      auth = '&gtm_auth=' + auth
+    }
+    if (preview) {
+      preview = '&gtm_preview=' + preview + '&gtm_cookies_win=x'
+    }
+
+    this.googleSrc = 'https://www.googletagmanager.com/gtm.js?id=' + window.GOVUK.analyticsGa4.vars.id + auth + preview
+    newScript.src = this.googleSrc
+    firstScript.parentNode.insertBefore(newScript, firstScript)
+  },
+
   loadGa4: function (currentDomain) {
```

Comment:
Again this code isn't part of the PR but I can't seem to find where we're calling `loadGa4` with parameters. Do we need to ultimately pass a parameter here? Although I think I've probably misunderstood something so apologies if this comment is redundant.

Reply:
@JamesCGDS That parameter is for our tests, it's used to override the domain we're on so that we can test that the right environment variables are loaded in the tests. Otherwise we couldn't test this code as jasmine always runs on `localhost` so it would always load the developer environment variables.

Reply:
Cool understood, thanks @AshGDS. Do you think there'd be any mileage in putting a quick comment here saying as such? (Sorry I know this isn't part of the PR)